### PR TITLE
RavenDB-14743: Add TransferEncodingChunked = true to smuggler HttpReq…

### PR DESCRIPTION
**Pull request by garayx**
…uest, propagate smuggler task to operation

### CODE REVIEW

File: src/Raven.Client/Documents/Operations/Operation.cs [hunk 1]
```diff
@@ -246,6 +246,12 @@ public void OnNext(OperationStatusChange change)
                         break;
                     case OperationStatus.Faulted:
                         StopProcessing();
+                        if (_additionalTask.IsFaulted)
+                        {
+                            _result.TrySetException(_additionalTask.Exception.ExtractSingleInnerException());
```
Comment (ayende) [inline src/Raven.Client/Documents/Operations/Operation.cs]:
```suggestion
                            _result.TrySetException(_additionalTask.Exception);
```

`.ExtractSingleInnerException()` should be on the caller, not here. We don't want to possibly lose information for the caller

File: test/EmbeddedTests/SmugglerTests.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                    using (store.SetRequestTimeout(TimeSpan.FromMilliseconds(1000), databaseName))
+                    {
+                        try
+                        {
+                            var operation = await store.Smuggler.ForDatabase(databaseName).ImportAsync(new DatabaseSmugglerImportOptions
+                            {
+                                OperateOnTypes = DatabaseItemType.Documents | DatabaseItemType.Identities | DatabaseItemType.CompareExchange,
+                            }, "dump.cmp");
+
+                            await operation.WaitForCompletionAsync();
```
Comment (ppekrol) [inline test/EmbeddedTests/SmugglerTests.cs:62]: You are not failing the test here. After this line you should add an assert or throw that it should not happen

File: test/EmbeddedTests/SmugglerTests.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            using (var embedded = new EmbeddedServer())
+            {
+                embedded.StartServer(new ServerOptions
+                {
+                    ServerDirectory = paths.ServerDirectory,
+                    DataDirectory = paths.DataDirectory,
+                });
+
+                const string databaseName = "test";
+                var dummyDump = CreateDummyDump(100000);
```
Comment (ppekrol) [inline test/EmbeddedTests/SmugglerTests.cs]: Is there a smaller version of this test available?

File: src/Raven.Client/Documents/Operations/Operation.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Operations/Operation.cs]: Have you pushed this?

Comment (garayx) [reply src/Raven.Client/Documents/Operations/Operation.cs]: yes

**Review by ppekrol — APPROVED** [review]

**Pull request merged**